lib/puppet/provider/gnupg_key/gnupg.rb

```diff
@@ -73,7 +79,7 @@ def add_key_from_key_content
     path = create_temporary_file(user_id, resource[:key_content])
     command = "gpg --import #{path}"
     begin
-      output = Puppet::Util::Execution.execute(command, :uid => user_id, :failonfail => true)
+      `#{command}`
```

Comment:
Yeah, you need to catch the returned error code and raise it as a puppet error i.e.

```ruby
code = `foo`
raise(Puppet::Error, "Failed with code #{code}") unless code == "0"